### PR TITLE
Feature: Limit and Paginate

### PR DESCRIPTION
**Pull request by poppabear8883**
Example: `api/users?limit=10&paginate=2`

```json
{
  "data": [
    {
      "id": 1,
      "email": "jeremy95@example.com",
      "account": {
        "email": "ykoss@thiel.org",
        "first_name": null,
        "last_name": null,
        "address_one": null,
        "address_two": null,
        "city": null,
        "territory": null,
        "country_code": null,
        "country": null,
        "postal_code": null,
        "phone": null,
        "alt_phone": null
      },
      "permissions": null,
      "last_login": null
    },
    {
      "id": 2,
      "email": "dana48@example.com",
      "account": {
        "email": "lindsey.bechtelar@legros.com",
        "first_name": null,
        "last_name": null,
        "address_one": null,
        "address_two": null,
        "city": null,
        "territory": null,
        "country_code": null,
        "country": null,
        "postal_code": null,
        "phone": null,
        "alt_phone": null
      },
      "permissions": null,
      "last_login": null
    }
  ],
  "links": {
    "first": "\/?page=1",
    "last": "\/?page=5",
    "prev": null,
    "next": "\/?page=2"
  },
  "meta": {
    "current_page": 1,
    "from": 1,
    "last_page": 5,
    "path": "\/",
    "per_page": 2,
    "to": 2,
    "total": 10
  }
}
```

### CODE REVIEW

**Comment by poppabear8883** [comment]
Thinking about this a bit more, I think I will refactor the way the tests are ... The test name doesn;t really reflect whats being tested here ...

**Comment by cbj4074** [comment]
Okay, I'll hold-off on a Review, then.

File: app/Http/Controllers/Api/ApiController.php [hunk 1]
```diff
@@ -59,12 +59,37 @@ public function __construct()
...
      */
-    public function all()
+    public function index(Request $request)
     {
+        $this->validate($request, [
+            'paginate' => 'numeric|min:1|max:100',
+            'limit' => 'numeric|min:1'
+        ]);
+
+        $has_paginate = $request->has('paginate');
```
[the remark this answers is not in the document]
Comment (cbj4074) [reply app/Http/Controllers/Api/ApiController.php]: I went ahead and fixed these.

[the remark this answers is not in the document]
Comment (poppabear8883) [reply app/Http/Controllers/Api/ApiController.php]: Yep you are correct ... good catch

**Pull request merged**